### DISCUSSION TITLE
Callbacks in command (Loading Sub Commands )

### DISCUSSION THREAD

**Discussion by BurnedChris** [asabaylus/react-command-palette#726] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
Hi, I recently added this to my project and its awesome but there a few things i am wondering how to do

**Describe the solution you'd like**
Feature 1) Be able to integrate a search into it in steps 
1) clicks search Users 
2) then allows to type into the command pallet a user name
3) that then runs an api command
4) builds new list and allows to select from user
5) navigate to user page

Feature 2) Be able to return a call back / promise in the function to close the loading indicator and also maybe even the modal,

as i have some almost instant commands that still leaves the spinner spinning for a few seconds after it!

**Describe alternatives you've considered**
For it to be more of a custom solution but the steps on how to get there considered, 

Maybe they could be a submenu built in but also a promise to fetch more submenus from an api? 

Thanks

**Comment by asabaylus** [comment, marked as answer]
@Burnsy it sounds like you want "sub-commands". There's an example in this codesandbox
https://codesandbox.io/s/react-command-palette-subcommands-gw1zy

The React Command Palette renders whatever list of commands it's passed at the time they are passed in. Using the `closeOnSelect` prop, the UI can be forced to remain open while the new commands are loaded. Provided the search is quick, under say 1-4 sec the user experience will be good. 

When the user selects a child command the spinner can be displayed using the `showSpinnerOnSelect` prop.

Please let me know if you have any additional questions.
